coral/src/domain/connector/connector-api.ts

```diff
@@ -39,25 +40,19 @@ const filterGetConnectorRequestParams = (
   });
 };
 
-const getConnectors = ({
-  currentPage,
-  environment = "ALL",
-  teamName,
-  connectorName,
-}: {
-  currentPage: number;
-  environment: string;
-  teamName?: string;
-  connectorName?: string;
-}) => {
-  const queryParams: KlawApiRequestQueryParameters<"getConnectors"> = {
-    pageNo: currentPage.toString(),
-    env: environment,
-    ...(teamName && { teamName: teamName }),
-    ...(connectorName && {
-      connectornamesearch: connectorName,
-    }),
-  };
+const getConnectors = (
+  params: KlawApiRequestQueryParameters<"getConnectors">
```

Comment:
I changed the argument to be explicit the right type. That way, there needs to be more handling of that in the component (which I don't really like, because the component should not care or even know how the API needs it data). But it enables a stronger typing - that's what called the last break on `main` -> `getTopics` took this arguments: 

```
{
  currentPage: number;
  environment: string;
  teamName: Team;
  searchTerm?: string; 
 }
 ```
 and then build the query, defined as a `Record<string, string>` and used that as URLSearchParams. 
 
When the endpoint `getTopics` changed to take `teamId` instead of `teamName`, TS compiler didn't have anything to complain here. 

This solution is not 100% safe either, but more sage at least. I'll make a follow up ticket for that, I'm sure we can find a way to make this more typesafe while removing unneeded logic from the component.